File: 0704-binary-search/0704-binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int s=0,e=nums.size()-1,m;
        while(s<=e) {
            m = (s+e)/2;
            if(nums[m]==target) return m;
            else if(target<nums[m]) e=m-1;
            else s=m+1;
        }
        return -1;
    }
};

File: 0704-binary-search/0704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        s=0
        e=len(nums)-1
        m=0
        while s<=e:
            m=int((s+e)/2)
            if nums[m]==target: return m
            elif nums[m]<target: s=m+1
            else: e=m-1
        return -1
        